### PR TITLE
Copter/Plane: Refactoring

### PR DESCRIPTION
**Pull request by muramura**
1. remove unnecessary preprocessors.
　　The same preprocessor is also used in the method.
2. One step can be deleted by making it a preprocessor.

### CODE REVIEW

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -659,11 +657,11 @@ bool AP_Arming_Copter::mandatory_checks(bool display_failure)
     AP_Notify::flags.pre_arm_gps_check = result;
 
     // call mandatory alt check
-    if (!alt_checks(display_failure)) {
+    if (result && !alt_checks(display_failure)) {
         result = false;
     }
 
-    return result & AP_Arming::mandatory_checks(display_failure);
+    return result && AP_Arming::mandatory_checks(display_failure);
```
Comment (IamPete1) [inline ArduCopter/AP_Arming.cpp:664]: this change means that we will not call all checks, lazy evaluation will skip `alt_checks` and `AP_Arming::mandatory_checks` if result is false. That means we don't get all the info on the first attempt.

File: ArduCopter/radio.cpp [hunk 2]
```diff
@@ -143,15 +143,13 @@ void Copter::set_throttle_and_failsafe(uint16_t throttle_pwm)
 
         // check for 3 low throttle values
         // Note: we do not pass through the low throttle until 3 low throttle values are received
-        failsafe.radio_counter++;
-        if( failsafe.radio_counter >= FS_COUNTER ) {
+        if( ++failsafe.radio_counter >= FS_COUNTER ) {
```
Comment (IamPete1) [inline ArduCopter/radio.cpp:146]: I think this pattern is harder to understand.

Comment (rmackay9) [reply ArduCopter/radio.cpp:146]: I agree, I'm not a fan of this change I'm afraid.

**Review by khancyr — COMMENTED** [review]
The changes aren't great : 
We need to keep the AP_AIRSPEED_ENABLED
We want to keep bitwise operator for arming check to ensure all checks are run
pre-increment/decrement makes reading harder here and don't bring speed benefit as the compiler already optimise this out.

Therefore, we will close this unless you want to point out some improvement we can have missed out.
Thanks anyway

**Pull request closed without merging**